collect_app/src/main/java/org/odk/collect/android/widgets/QuestionWidget.java

```diff
@@ -111,34 +112,33 @@ public QuestionWidget(Context context, FormEntryPrompt prompt) {
         textLayout = helpTextLayout.findViewById(R.id.text_layout);
         warningText = helpTextLayout.findViewById(R.id.warning_text);
         helpTextView = setupHelpText(helpTextLayout.findViewById(R.id.help_text_view), prompt);
-        guidanceTextView = setupGuidanceTextAndLayout(helpTextLayout.findViewById(R.id.guidance_text_view), prompt);
+        setupGuidanceTextAndLayout(helpTextLayout.findViewById(R.id.guidance_text_view), prompt);
 
         addQuestionMediaLayout(getQuestionMediaLayout());
         addHelpTextLayout(getHelpTextLayout());
     }
 
-    private TextView setupGuidanceTextAndLayout(TextView guidanceTextView, FormEntryPrompt prompt) {
+    private void setupGuidanceTextAndLayout(TextView guidanceTextView, FormEntryPrompt prompt) {
 
-        TextView guidance = null;
         GuidanceHint setting = GuidanceHint.get((String) GeneralSharedPreferences.getInstance().get(GeneralKeys.KEY_GUIDANCE_HINT));
 
         if (setting.equals(GuidanceHint.No)) {
-            return null;
+            return;
         }
 
         String guidanceHint = prompt.getSpecialFormQuestionText(prompt.getQuestion().getHelpTextID(), "guidance");
 
-        if (android.text.TextUtils.isEmpty(guidanceHint)) {
-            return null;
+        if (isEmpty(guidanceHint)) {
+            return;
         }
 
-        guidance = configureGuidanceTextView(guidanceTextView, guidanceHint);
+        configureGuidanceTextView(guidanceTextView, guidanceHint);
 
         expanded = new AtomicBoolean(false);
 
         if (getState() != null) {
             if (getState().containsKey(GUIDANCE_EXPANDED_STATE + getFormEntryPrompt().getIndex())) {
-                Boolean result = getState().getBoolean(GUIDANCE_EXPANDED_STATE + getFormEntryPrompt().getIndex());
+                boolean result = getState().getBoolean(GUIDANCE_EXPANDED_STATE + getFormEntryPrompt().getIndex());
                 expanded = new AtomicBoolean(result);
```

Comment:
I’m suspicious of an AtomicBoolean that’s recreated rather than set. I didn’t study this closely though.